### PR TITLE
fix tests with MOI 1.6

### PR DESCRIPTION
**Pull request by frapac**

### CODE REVIEW

File: test/MOI_wrapper.jl [hunk 1]
```diff
@@ -55,6 +55,7 @@ const MOI_BASE_EXCLUDED = String[
     "test_linear_integration_Interval", # TODO ? Knitro 13.0 converges to 9.99999* instead of 10.0
     # QUADRATIC
     "test_quadratic_Integer_SecondOrderCone", # MOI.get(model, MOI.TerminationStatus()) == MOI.OTHER_LIMIT
+    "test_cpsat",
```
Comment (odow) [inline test/MOI_wrapper.jl:58]: Do these fail? There should be MILP reformulations.

Comment (odow) [reply test/MOI_wrapper.jl:58]: Hmm. I thought I fixed those. Let me double check.

**Comment by odow** [comment]
Oh. There's no CI?

File: test/MOI_wrapper.jl (hunk 1 above)
Comment (odow) [reply test/MOI_wrapper.jl:58]: What's the full error?

**Comment by frapac** [comment]
Unfortunately, no ... we don't have a license to run the CI independently... 

The full error I get is: 
```
 Unable to use CountDistinctToMILPBridge because element 2 in the function has a non-finite domain: MathOptInterface.VariableIndex(1)                                                         
  Stacktrace:                                                                                                                                                                                  
    [1] error(s::String)                                                                                                                                                                       
      @ Base ./error.jl:33                                                                                                                                                                     
    [2] final_touch(bridge::MathOptInterface.Bridges.Constraint.CountDistinctToMILPBridge{Float64, MathOptInterface.VectorOfVariables}, model::MathOptInterface.Bridges.LazyBridgeOptimizer{KNI
TRO.Optimizer})                                                                                                                                                                                
      @ MathOptInterface.Bridges.Constraint ~/.julia/packages/MathOptInterface/AiEiQ/src/Bridges/Constraint/bridges/count_distinct.jl:278                                                      
    [3] _final_touch(bridges::OrderedCollections.OrderedSet{MathOptInterface.Bridges.Constraint.CountDistinctToMILPBridge{Float64, MathOptInterface.VectorOfVariables}}, model::MathOptInterfac
e.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer})                                                                                                                                               
      @ MathOptInterface.Bridges.Constraint ~/.julia/packages/MathOptInterface/AiEiQ/src/Bridges/Constraint/map.jl:324                                                                         
    [4] final_touch(map::MathOptInterface.Bridges.Constraint.Map, model::MathOptInterface.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer})                                                       
      @ MathOptInterface.Bridges.Constraint ~/.julia/packages/MathOptInterface/AiEiQ/src/Bridges/Constraint/map.jl:331                                                                         
    [5] final_touch                                                                                                                                                                            
      @ ~/.julia/packages/MathOptInterface/AiEiQ/src/Bridges/bridge_optimizer.jl:462 [inlined]                                                                                                 
    [6] optimize!                                                                                                                                                                              
      @ ~/.julia/packages/MathOptInterface/AiEiQ/src/Bridges/bridge_optimizer.jl:378 [inlined]                                                                                                 
    [7] test_cpsat_AllDifferent(model::MathOptInterface.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer}, config::MathOptInterface.Test.Config{Float64})                                          
      @ MathOptInterface.Test ~/.julia/packages/MathOptInterface/AiEiQ/src/Test/test_cpsat.jl:27                                                                                               
    [8] macro expansion                                                                                                                                                                        
      @ ~/.julia/packages/MathOptInterface/AiEiQ/src/Test/Test.jl:220 [inlined]                                                                                                                
    [9] macro expansion                                                                                                                                                                        
      @ ~/packages/julia-1.7.2/share/julia/stdlib/v1.7/Test/src/Test.jl:1283 [inlined]                                                                                                         
   [10] runtests(model::MathOptInterface.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer}, config::MathOptInterface.Test.Config{Float64}; include::Vector{String}, exclude::Vector{String}, warn_u
nsupported::Bool, exclude_tests_after::VersionNumber)                                                                                                                                          
      @ MathOptInterface.Test ~/.julia/packages/MathOptInterface/AiEiQ/src/Test/Test.jl:215                                                                                                    
   [11] test_MOI_Test_bridged()                                                                                                                                                                
      @ Main.TestMOIWrapper ~/dev/KNITRO.jl/test/MOI_wrapper.jl:95                                                                                                                             
   [12] macro expansion                                                                                                                                                                        
      @ ~/dev/KNITRO.jl/test/MOI_wrapper.jl:12 [inlined]                                                                                                                                       
   [13] macro expansion                                                                                                                                                                        
      @ ~/packages/julia-1.7.2/share/julia/stdlib/v1.7/Test/src/Test.jl:1283 [inlined]                                                                                                         
   [14] runtests()                                                                                                                                                                             
      @ Main.TestMOIWrapper ~/dev/KNITRO.jl/test/MOI_wrapper.jl:12                                                                                                                             
   [15] top-level scope                                                                                                                                                                        
      @ ~/dev/KNITRO.jl/test/MOI_wrapper.jl:101    
```

**Comment by odow** [comment]
Hmm. Something is wrong. Because the variables don't have finite domain:
https://github.com/jump-dev/MathOptInterface.jl/blob/bcc869f09bc3916683c6d296237ba5bfe25aa6f9/src/Test/test_cpsat.jl#L25

MILP solvers like HiGHS and Gurobi pass the tests. I wonder if there's a problem in KNITRO querying variable bounds.

**Comment by odow** [comment]
The fact that these are excluded makes me wonder if there is a wider part of the API missing:
https://github.com/jump-dev/KNITRO.jl/blob/e7be7938a1cbb95d933876b3110aa66b347d1409/test/MOI_wrapper.jl#L27-L29

**Comment by frapac** [comment]
If Gurobi and Highs passed the test, I am fairly confident that `MOI.Tests` is correct.

I confirm that part of the API is indeed missing. When the MOI wrapper was implemented in 2018, Knitro API didn't support the operations needed to implement `ObjectiveFunction` and `ConstraintFunction`, and since then the implementation remains incomplete. I think we should address this issue now. I will work on that during the week.

**Comment by odow** [comment]
I'm not really sure how it's passing tests, but my guess is that the excludes skip a bunch unintentionally. 

For example, `ListOfConstraintIndices` isn't implemented:
https://github.com/jump-dev/KNITRO.jl/search?q=ListOfConstraintIndices

I'd remove all the excludes, go through each test, and then only exclude a test if you can pin-point exactly why you want to skip it (i.e., it tests an infeasibility certificate).

**Pull request closed without merging**